internal/common/common_test.go

```diff
@@ -466,6 +466,55 @@ var _ = Describe("JSON serialization checks", func() {
 	})
 })
 
+var _ = Describe("Test CanUnbindhost", func() {
+	Context("Verify Host cannot be removed while cluster progressing", func() {
+		hosts := make([]*models.Host, 0)
+		var cluster Cluster
+		BeforeEach(func() {
+			for i := 1; i < 5; i++ {
+				// Creating full cluster 3 masters 3 nodes
+				hosts = append(hosts, createHost(models.HostRoleMaster, models.HostStatusKnown))
+				hosts = append(hosts, createHost(models.HostRoleWorker, models.HostStatusKnown))
+			}
+			cluster = Cluster{
+				Cluster: models.Cluster{
+					Hosts: hosts,
+				},
+			}
+		})
+		It("fail while cluster is progressing", func() {
+			notAllowedStatus := []string{
+				models.ClusterStatusFinalizing,
+				models.ClusterStatusInstalling,
+				models.ClusterStatusInstallingPendingUserAction,
+				models.ClusterStatusPreparingForInstallation,
+			}
+			for _, failState := range notAllowedStatus {
+				cluster.Status = swag.String(failState)
+				err := CanUnbindhost(&cluster)
+
+				Expect(err).ToNot(BeNil())
```

Comment:
Whereas this checks that there is error handling generally it doesn't verify that the error here https://github.com/openshift/assisted-service/pull/5809/files#diff-e0a57b903e60761f673a4485b4a747f500f4155828e182cc19786ee5648cd6e6R448 was the one emitted. 

I'd personally update the test to verify that the expected error is returned.

Reply:
correct, it does verify the exact error, just check if we go err back.
I'll added also the error string check 
thanks